opensearch-service-domain-cdk/lib/stack-composer.ts

```diff
@@ -0,0 +1,181 @@
+        const customEndpointCertificateARN = getContextForType('customEndpointCertificateARN', 'string')
+        const customEndpointHostedZoneId = getContextForType('customEndpointHostedZoneId', 'string')
+        const enforceHTTPS = getContextForType('enforceHTTPS', 'boolean')
+        const noneToNodeEncryptionEnabled = getContextForType('nodeToNodeEncryptionEnabled', 'boolean')
+        const encryptionAtRestEnabled = getContextForType('encryptionAtRestEnabled', 'boolean')
+        const encryptionAtRestKmsKeyARN = getContextForType("encryptionAtRestKmsKeyARN", 'string')
+        const ebsEnabled = getContextForType('ebsEnabled', 'boolean')
+        const ebsIops = getContextForType('ebsIops', 'number')
+        const ebsVolumeSize = getContextForType('ebsVolumeSize', 'number')
+        const loggingAppLogEnabled = getContextForType('loggingAppLogEnabled', 'boolean')
+        const loggingAppLogGroupARN = getContextForType('loggingAppLogGroupARN', 'string')
+        const loggingAuditLogEnabled = getContextForType('loggingAuditLogEnabled', 'boolean')
+        const loggingAuditLogGroupARN = getContextForType('loggingAuditLogGroupARN', 'string')
+        const loggingSlowIndexLogEnabled = getContextForType('loggingSlowIndexLogEnabled', 'boolean')
+        const loggingSlowIndexLogGroupARN = getContextForType('loggingSlowIndexLogGroupARN', 'string')
+        const loggingSlowSearchLogEnabled = getContextForType('loggingSlowSearchLogEnabled', 'boolean')
+        const loggingSlowSearchLogGroupARN = getContextForType('loggingSlowSearchLogGroupARN', 'string')
+        const snapshotAutomatedStartHour = getContextForType('snapshotAutomatedStartHour', 'number')
+        const vpcId = getContextForType('vpcId', 'string')
+        const vpcSecurityGroupIds = getContextForType('vpcSecurityGroupIds', 'object')
+        const vpcSubnetIds = getContextForType('vpcSubnetIds', 'object')
+        const enableVersionUpgrade = getContextForType('enableVersionUpgrade', 'boolean')
+
+        // Check for partially configured features
+        if ((cognitoIdentityPoolId || cognitoRoleARN || cognitoUserPoolId) &&
+            !(cognitoIdentityPoolId && cognitoRoleARN && cognitoUserPoolId)) {
+            throw new Error("All Cognito values [cognitoIdentityPoolId, cognitoRoleARN, cognitoUserPoolId] are " +
+                "required if using Cognito authentication")
+        }
+
+        if ((customEndpointCertificateARN || customEndpointHostedZoneId) && !(customEndpointDomainName)) {
+            throw new Error("The 'customEndpointDomainName' value is required when configuring a custom endpoint")
+        }
+
+        const engineVersion = getContextForType('engineVersion', 'string')
+        if (engineVersion.startsWith("OS_")) {
+            // Will accept a period delimited version string (i.e. 1.3) and return a proper EngineVersion
+            version = EngineVersion.openSearch(engineVersion.substring(3))
```

Comment:
what happens if I pass in something undefined like OS_99.99?

Reply:
I think that's probably the behavior we want